airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -469,23 +454,61 @@ private <T> T getWorkflowOptionsWithWorkflowId(final Class<T> workflowClass, fin
     return client.newWorkflowStub(workflowClass, TemporalUtils.getWorkflowOptionsWithWorkflowId(jobType, name));
   }
 
-  private <T> T getExistingWorkflow(final Class<T> workflowClass, final String name) {
-    return client.newWorkflowStub(workflowClass, name);
-  }
-
-  ConnectionManagerWorkflow getConnectionUpdateWorkflow(final UUID connectionId) {
-    final boolean workflowReachable = isWorkflowReachable(getConnectionManagerName(connectionId));
-
-    if (!workflowReachable) {
-      throw new IllegalStateException("No reachable workflow for the connection {} while trying to delete it");
+  /**
+   * Attempts to retrieve the connection manager workflow for the provided connection.
+   *
+   * @param connectionId the ID of the connection whose workflow should be retrieved
+   * @return the healthy ConnectionManagerWorkflow
+   * @throws DeletedWorkflowException if the workflow was deleted, according to the workflow state
+   * @throws UnreachableWorkflowException if the workflow is unreachable
+   */
+  private ConnectionManagerWorkflow getConnectionManagerWorkflow(final UUID connectionId)
```

Comment:
This method and the following method are the main changes in this PR - these methods offer the rest of this class a way to retrieve the connection manager workflow while also enforcing that they handle the deleted case and the unreachable workflow case through exceptions. The second method `repairAndRetrieveConnectionManagerWorkflow` handles the unreachable case automatically by automatically restarting the temporal workflow.